plugins/serializers/binary/entry.go

```diff
@@ -0,0 +1,287 @@
+}
+
+func appendToSlice[T any](arr []T, targetLength uint64, value T) []T {
+	for uint64(len(arr)) < targetLength {
+		arr = append(arr, value)
+	}
+
+	return arr
+}
+
+func (e *Entry) SerializeValue(value interface{}, converter binary.ByteOrder) ([]byte, error) {
+	if e.DataFormat != "string" {
+		// time to int64
+		switch v := value.(type) {
+		case time.Time:
+			value = e.timeToTimestamp(v)
+		case string:
+			var err error
+			value, err = strconv.ParseFloat(v, 64)
+			if err != nil {
+				return nil, fmt.Errorf("string to float conversion failed: %w", err)
+			}
+		}
+	}
+
+	// to any of int64, int32, int16, int8, int, uint64, uint32, uint16, uint8, uint types
+	if strings.HasPrefix(e.DataFormat, "int") || strings.HasPrefix(e.DataFormat, "uint") {
+		v := reflect.ValueOf(value)
+		v = reflect.Indirect(v)
+
+		switch value.(type) {
+		case int64, int32, int16, int8, int:
+			// check if overflow/data loss is possible
+			in := v.Convert(reflect.TypeOf(int64(0))).Int()
+			isIntegerConversionPossible(in, e.DataFormat, e.bytes)
+
+			// convert to target type
+			uin := v.Convert(getCorrespondingUintType(e.DataFormat)).Uint()
+			bytes := make([]byte, 8)
+			converter.PutUint64(bytes, uin)
+
+			return trimByteSliceByEndianness(bytes, e.bytes, converter), nil
+		case uint64, uint32, uint16, uint8, uint:
+			// check if overflow/data loss is possible
+			in := v.Convert(reflect.TypeOf(uint64(0))).Uint()
+			isUIntegerConversionPossible(in, e.DataFormat, e.bytes)
+
+			// convert to target type
+			uin := v.Convert(getCorrespondingUintType(e.DataFormat)).Uint()
+			bytes := make([]byte, 8)
+			converter.PutUint64(bytes, uin)
+
+			return trimByteSliceByEndianness(bytes, e.bytes, converter), nil
+		case float32, float64:
+			// check if overflow/data loss is possible
+			log.Printf("I! [binary] float to integer conversion detected. Loss of data is very likely")
+			uin := v.Convert(getCorrespondingUintType(e.DataFormat)).Uint()
+			bytes := make([]byte, 8)
+			converter.PutUint64(bytes, uin)
+
+			return trimByteSliceByEndianness(bytes, e.bytes, converter), nil
+		}
+	}
+
+	// to float32, float64
+	if e.DataFormat == "float32" || e.DataFormat == "float64" {
+		v := reflect.ValueOf(value)
+		v = reflect.Indirect(v)
+		f := v.Convert(reflect.TypeOf(float64(0))).Float()
+
+		isFloatConversionPossible(f, e.DataFormat)
+
+		switch e.DataFormat {
+		case "float32":
+			bytes := make([]byte, 4)
+			bits32 := math.Float32bits(float32(f))
+			converter.PutUint32(bytes, bits32)
+
+			return bytes, nil
+		case "float64":
+			bytes := make([]byte, 8)
+			bits64 := math.Float64bits(f)
+			converter.PutUint64(bytes, bits64)
+
+			return bytes, nil
+		}
+	}
+
+	// to string
+	if e.DataFormat == "string" {
+		str := fmt.Sprintf("%v", value)
+		// str to bytes
+		bytes := []byte(str)
+		if len(bytes) >= int(e.bytes) {
+			return append(bytes[:int(e.bytes)-1], e.termination), nil
+		}
+
+		return appendToSlice(bytes, e.bytes, e.termination), nil
```

Comment:
How about
```suggestion
		s, err := internal.ToString(value)
		if err != nil {
			return nil, err
		}

		buf := []bytes(s)
		buflen := len(buf)
		if buflen >= int(e.bytes) {
			return buf[:e.bytes], nil
		}
		buf = append(buf, e.termination)
		buflen++
		
		if buflen < int(e.bytes) {
			pad := make([]buf, int(e.bytes)-buflen)
			buf = append(buf, pad...)
		}

		return buf, nil
```

Reply:
Hm, I'm not sure about the change in `line 215` as it changes the logic pretty much. I think termination byte is more important than an extra byte of data when receiver is written e.g. in C / C++ and expects strings to be null terminated. 

E.g. when string is `123456`  and we expect `5` bytes (chars)  I think it is more important to send `1234\0` than `12345`. This way client can always expect strings to be correctly terminated and string functions work as expected.

@srebhan What do You think?

Reply:
I'm fine with shorten the string and add the terminator. Just make this behavior clear in the config-comment.